### ISSUE TITLE
Three double slashes an equal-length copy cannot spell out of harm's way

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#216]
Found by an adversarial review of the branch that repairs #153, and left standing by it deliberately: these are the shapes an equal-length copy cannot answer, so they were kept out of that repair rather than answered badly.

`hideFalseInlineComments` hands `style-search`, and the pattern inside `functionCommaSpaceChecker`, a copy of the text in which the second slash of every double slash that opens no comment is written as a hyphen. The copy is as long as the text and spells it character for character everywhere else, so every position stands where it did. Two shapes fall outside what such a copy can say.

### What minimal example or steps are needed to reproduce the bug?

```css
a { b: 1px/*x*//*y*/,2px; }
a { b: image-set(url(a//*b*/) 1x ,url(//c) 2x); }
a { b: myurl(//a),2px; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": "@stylistic/stylelint-plugin",
  "rules": {
    "@stylistic/value-list-comma-space-before": "always",
    "@stylistic/function-comma-space-before": "never"
  }
}
```

### What did you expect to happen?

A warning about the comma of the first declaration, which has nothing in front of it; nothing about the comma of the second, which has the space `never` forbids in front of it and should therefore be reported once the arguments are read correctly — and no warning at all where the file already reads as it should; a warning about the comma of the third.

### What actually happened?

Nothing about the first. The `*/` closing one comment and the `/*` opening the next spell `//` between them, which `style-search` reads as an inline comment opening; a hyphen cannot be written over either slash, since each is half of a delimiter the reader needs, so the copy leaves the pair as it stands. `findSelectorInlineComments` answers the same shape in a selector — that was #164 — by putting a space between the two comments, which a selector can carry and a value cannot.

The second is reported where it should be silent, and silent where it should be reported. `url(a//*b*/)` holds a double slash whose second slash opens a block comment, so the hyphen is held back there too, and `COMMENT_PATTERN` inside `functionCommaSpaceChecker` then reads from `url(a` to the end of the arguments as one comment. Masking the first slash instead would answer this one — unless that first slash is itself the `/` of a preceding `*/`, which is the shape above.

Nothing about the third, and this in plain CSS, where no comment of that kind exists at all. `findInlineCommentSpans` reads a `url()` token by its name, and its name has to stand on its own: `image-url(` and `@{prefix}url(` end in the same four characters while being ordinary calls whose arguments may hold a comment. A double slash inside `myurl(…)` is therefore an inline comment to it, as it is to the search — while in a plain CSS file it is two literal slashes and nothing more. The reading a syntax asks for is what is missing here: a file of plain CSS has no inline comments to find, and every double slash in one is text.

### Do you have a proposal to fix the bug?

For the last of the three: let `findInlineCommentSpans` be told which syntax spelled the text, and return no span at all for a text of plain CSS. Every caller already knows — the rules are handed the syntax through `postcss`, and `endsInlineCommentOnFormFeed` already asks the node a related question.

For the first two: an equal-length copy is the wrong instrument. Either the search learns to read a text the way `findInlineCommentSpans` reads it — a reader rather than a scan of characters — or the callers stop handing it a copy and place their matches themselves.
